components/core/src/clp/streaming_archive/MetadataDB.hpp

```diff
@@ -75,6 +75,7 @@ class MetadataDB {
                 epochtime_t begin_timestamp,
                 epochtime_t end_timestamp,
                 std::string const& file_path,
+                std::string const& file_id,
```

Comment:
we should use `std::optional<std::string_view>`:
- `std::nullopt` to indicate the value is not given
- `std::string_view` to avoid data copy

Reply:
The same applies to file_path, I would consider it to be another PR